### PR TITLE
Include cuDNN function name in error message

### PR DESCRIPTION
**Pull request by niboshi**
cuDNN functions tend to be difficult to debug.
This PR includes cuDNN function names in error messages.

### CODE REVIEW

**Comment by hvy** [comment]
What is the status of this PR?

**Comment by niboshi** [comment]
Rebased and removed WIP. PTAL

**Comment by niboshi** [comment]
@hvy Can you review?

File: chainerx_cc/chainerx/cuda/cudnn.h [hunk 1]
```diff
@@ -10,21 +12,40 @@
 #include "chainerx/macro.h"
 #include "chainerx/stack_vector.h"
 
+#define CHAINERX_CUDA_CUDNN_CALL_WITH_HANDLE(handle, func, args...)                                                                    \
```
Comment (hvy) [inline chainerx_cc/chainerx/cuda/cudnn.h]: Actually, I don't think we need this macro? We can simply use `CHAINERX_CUDA_CUDNN_CALL` passing the handle similar to any other argument to the cuDNN routine. It's easier from the caller perspective I think.

File: chainerx_cc/chainerx/cuda/cudnn.h [hunk 2]
```diff
@@ -10,21 +12,40 @@
...
     cudnnStatus_t error() const noexcept { return status_; }
 
 private:
     cudnnStatus_t status_{};
 };
 
-void CheckCudnnError(cudnnStatus_t status);
+namespace cudnn_detail {
+
+inline void CheckCudnnError(const char* func_name, cudnnStatus_t status) {
```
Comment (hvy) [inline chainerx_cc/chainerx/cuda/cudnn.h]: A detail but how about passing `func_name` as the later argument as it's less significant (in most cases never used).

**Comment by hvy** [comment]
Sorry for the delay, I wrote some comments. It's worth noting that we might want similar features for cuBLAS and soon cuSOLVER?

File: chainerx_cc/chainerx/cuda/cudnn.h (hunk 1 above)
Comment (niboshi) [reply chainerx_cc/chainerx/cuda/cudnn.h]: I don't think `CudnnHandle::Call` can be bypassed.
It has a concurrency control.

File: chainerx_cc/chainerx/cuda/cudnn.h (hunk 2 above)
Comment (niboshi) [reply chainerx_cc/chainerx/cuda/cudnn.h]: What do you mean by never used? It's mandatory.

Comment (hvy) [reply chainerx_cc/chainerx/cuda/cudnn.h]: It's only used when the condition evaluates to true, that's what I meant...

Comment (niboshi) [reply chainerx_cc/chainerx/cuda/cudnn.h]: Oh does that matter? `func_name` is more "stable" parameter, so I think it's natural to come first.

File: chainerx_cc/chainerx/cuda/cudnn.h (hunk 1 above)
Comment (hvy) [reply chainerx_cc/chainerx/cuda/cudnn.h]: Ah, that is true of course. Thanks.

File: chainerx_cc/chainerx/cuda/cudnn.h (hunk 2 above)
Comment (hvy) [reply chainerx_cc/chainerx/cuda/cudnn.h]: I thought so. Or at least, I'm not sure what you mean with "stable", but the intent of this functions is to check the status, doesn't `status` have a higher importance? 🤔

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Some segfaults? 
```
Aborted                 (core dumped) 
```

**Comment by emcastillo** [comment]
Let's close this, Thanks!

**Pull request closed without merging**